packages/back-end/src/controllers/experiments.ts

```diff
@@ -2591,7 +2658,17 @@ export async function deleteVisualChangeset(
     visualChangeset.experiment
   );
 
-  const envs = experiment ? getAffectedEnvsForExperiment({ experiment }) : [];
+  if (!experiment) {
+    throw new Error("Experiment not found");
+  }
```

Comment:
I don't think we want to block deleting a visual changeset if the experiment doesn't exist.  For example, this may be used to clean up orphaned documents.